pkgs/common-updater/scripts/list-directory-versions

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
+import argparse
+import requests
+import os
+import subprocess
+import re
+from bs4 import BeautifulSoup
+
+parser = argparse.ArgumentParser(description="lists all available versions listed for a package in a site.")
+
+os.environ.get('UPDATE_NIX_PNAME', '')
+
+parser.add_argument("--pname", default=os.environ.get('UPDATE_NIX_PNAME', ''), help="Name of the package.")
+parser.add_argument("--attr-path", default=os.environ.get('UPDATE_NIX_ATTR_PATH', ''), help="Attribute path of the package.")
+parser.add_argument("--url", help="Url of the page that lists the package versions.")
+parser.add_argument("--file", help="File name for writing debugging information.")
+
+if __name__ == "__main__":
+    args = parser.parse_args()
+
+    pname = getattr(args, "pname")
+    attr_path = getattr(args, "attr_path")
+    url = getattr(args, "url")
+    file = getattr(args, "file")
+
+    if not url:
+        url=subprocess.check_output(
+            ["nix-instantiate", "--eval", "-E",
+             f"with import ./. {{}}; dirOf (lib.head {attr_path}.src.urls)",
+             ],
+            text=True,
+        ).strip('"\n')
+
+    # print a debugging message
+    if file:
+        with open(file, "a") as f:
+            f.write(f"# Listing versions for {pname} from {url}\n")
+
+    page = requests.get(url)
+    soup = BeautifulSoup(page.content, "html.parser")
+    links = soup.find_all("a")
+    for link in links:
+        link_url = link["href"]
```

Comment:
`<a name="anchor-name" />` is somewhat common.

```suggestion
        link_url = link.get("href", None)
        if link_url is not None:
            ...
```

Reply:
Done.